src/core/execute.c

```diff
@@ -4618,6 +4608,28 @@ int exec_spawn(Unit *unit,
                 n_storage_fds = params->n_storage_fds;
         }
 
+        r = locate_executable(command->path, &executable);
```

Comment:
this needs to be done after the namespace magic in the child, since it would otherwise reference files from the host namespace, which would fall apart with RootDirectory=/RootImage=.

also, setuid() should be done in the child first, since binaries might not be accessible to the final user that are to root

Reply:
Right, thanks for the hint. I had to move some mac stuff around, but now everything is checked after the mount namespace has been arranged and that looks much better.